blockstore/splitstore/splitstore.go

```diff
@@ -529,6 +568,7 @@ func (s *SplitStore) Start(chain ChainAccessor) error {
 func (s *SplitStore) Close() error {
 	atomic.StoreInt32(&s.closing, 1)
 
+	// I'm _pretty_ sure we can just use a lock here now. Doesn't even need to be a rwmutex.
```

Comment:
it will also need a wait group to use instead of the poll. I don't feel strongly about this code and either way works. 
The polll does have the advantage that it spams the lock so that the operator will see it.
Other than that, lock+waitgroup is probably nicer.

I don't feel strongly about this, so up to you.